src/pages/EventCreationPage/components/EventCreationForm/index.tsx

```diff
@@ -0,0 +1,200 @@
+import React from 'react';
+import { Formik, Field, Form } from 'formik';
+import { TextField } from 'formik-material-ui';
+import { DateTimePicker } from 'formik-material-ui-pickers';
+import DateFnsUtils from '@date-io/date-fns';
+import { Button, LinearProgress, Grid } from '@material-ui/core';
+import { MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { formatISO } from 'date-fns';
+
+import schema from './schema';
+
+import {
+  OfficerNameAutocomplete,
+  EventTypeDropdownField,
+} from '@SharedComponents';
+import { OfficerNameData } from '@SharedComponents/autocomplete/OfficerNameAutocomplete';
+import { EventTypeEnum } from '@Services/EventService';
+import { EventRequest } from '@Services/api/models';
+
+interface InitialValuesType {
+  startDate: string;
+  endDate: string;
+  name: string;
+  location: string;
+  description: string;
+  fbURL: string;
+  canvaURL: string;
+  type: EventTypeEnum | undefined;
+  hosts: OfficerNameData[];
+}
+
+const INITIAL_VALUES: InitialValuesType = {
+  startDate: formatISO(new Date()),
+  endDate: formatISO(new Date()),
+  name: '',
+  type: EventTypeEnum.Social,
+  hosts: [],
+  location: '',
+  description: '',
+  fbURL: '',
+  canvaURL: '',
+};
+
+interface EventCreationFormProps {
+  handleSubmit: (
+    values: EventRequest,
+    setSubmitting: (_: boolean) => void
+  ) => void;
+  handleCancel: () => void;
+}
+
+export const EventCreationForm = (props: EventCreationFormProps) => {
+  const { handleSubmit, handleCancel } = props;
+  const urlObjects = [
+    {
+      name: 'fbURL',
+      label: 'Facebook URL',
+    },
+    {
+      name: 'canvaURL',
+      label: 'Canva URL',
+    },
+  ];
+
+  return (
+    <MuiPickersUtilsProvider utils={DateFnsUtils}>
+      <Formik
+        initialValues={INITIAL_VALUES}
+        validationSchema={schema}
+        onSubmit={(values, { setSubmitting }) => {
+          handleSubmit(values, setSubmitting);
+        }}
+      >
+        {({ submitForm, isSubmitting }) => (
```

Comment:
later extract out form into component to be reused - we should aim to extract out components sooner than later